pkg/agent/agent.go

```diff
@@ -642,14 +645,51 @@ func (i *Initializer) initNodeLocalConfig() error {
 		return err
 	}
 
+	var nodeIPAddr, transportIPAddr *net.IPNet
+	var localIntf *net.Interface
+	// Find the interface configured with Node IP and use it for Pod traffic.
 	ipAddr, err := k8s.GetNodeAddr(node)
 	if err != nil {
 		return fmt.Errorf("failed to obtain local IP address from K8s: %w", err)
 	}
-	localAddr, localIntf, err := getIPNetDeviceFromIP(ipAddr)
+	nodeIPAddr, localIntf, err = getIPNetDeviceFromIP(ipAddr)
 	if err != nil {
 		return fmt.Errorf("failed to get local IPNet device with IP %v: %v", ipAddr, err)
 	}
+	transportIPAddr = nodeIPAddr
+	if i.networkConfig.TransportIface != "" {
+		// Find the configured transport interface, and update its IP address in Node's annotation.
+		transportIPAddr, localIntf, err = getTransportIPNetDeviceByName(i.networkConfig.TransportIface, i.ovsBridge)
```

Comment:
Do you think we can wrap the code of updating Node annotation to be shared with other cases (e.g. Node MAC)? @tnqn